### PR TITLE
TX History v1

### PR DESCRIPTION
**Pull request by FrederikBolding**

### CODE REVIEW

**Comment by FrederikBolding** [comment]
[ch6733]

File: src/services/ApiService/History/types.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+  readonly txType: ITxType;
+}
+
+export interface ITxHistoryERC20Transfer {
+  readonly from: TAddress;
+  readonly to: TAddress;
+  readonly contractAddress: TAddress;
+  readonly amount: number;
+}
```
Comment (blurpesec) [inline src/services/ApiService/History/types.ts:27]: Some of these props are of type `number`. I'm surprised this doesn't toss any overflow/underflow issues with the BN-esque values the api returns. Is this a potential problem?

File: src/services/Store/StoreProvider.tsx [hunk 2]
```diff
@@ -337,6 +339,20 @@ export const StoreProvider: React.FC = ({ children }) => {
...
     }, {}) || {};
 
+  // TX HISTORY
+  const [txHistory, setTxHistory] = useState<ITxHistoryApiResponse[]>([]);
+
+  useEffectOnce(() => {
+    HistoryService.instance
+      .getHistory(accounts.filter((a) => a.networkId === DEFAULT_NETWORK).map((a) => a.address))
+      .then((history) => {
+        console.debug(history);
```
Comment (blurpesec) [inline src/services/Store/StoreProvider.tsx:349]: Remove this?

File: src/services/TxHistory/helpers.tsx [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+  };
+};
+
+export const merge = (apiTxs: ITxReceipt[], accountTxs: ITxReceipt[]): ITxReceipt[] => {
+  // Prioritize Account TX - needs to be more advanced?
+  const filteredApiTxs = apiTxs.filter((tx) => !accountTxs.find((a) => a.hash === tx.hash));
+  return filteredApiTxs.concat(accountTxs).sort((a, b) => (a.timestamp || 0) - (b.timestamp || 0));
+};
+
+export const deriveTxType = (accountsList: StoreAccount[], tx: ITxReceipt): ITxHistoryType => {
```
Comment (blurpesec) [inline src/services/TxHistory/helpers.tsx:41]: This isn't tested anywhere, right?

File: src/services/TxHistory/helpers.tsx [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+    gasUsed: bigNumberify(tx.gasUsed || 0),
+    value: parseEther(tx.value.toString()),
+    nonce: tx.nonce.toString()
+  };
+};
+
+export const merge = (apiTxs: ITxReceipt[], accountTxs: ITxReceipt[]): ITxReceipt[] => {
+  // Prioritize Account TX - needs to be more advanced?
+  const filteredApiTxs = apiTxs.filter((tx) => !accountTxs.find((a) => a.hash === tx.hash));
+  return filteredApiTxs.concat(accountTxs).sort((a, b) => (a.timestamp || 0) - (b.timestamp || 0));
```
Comment (blurpesec) [inline src/services/TxHistory/helpers.tsx:38]: I might be mistaken, but I believe the base table component already sorts the txs when a sortable column is specified. Do we need them to be available pre-sorted in the context for some other reason?

File: src/services/Store/StoreProvider.tsx (hunk 2 above)
Comment (FrederikBolding) [reply src/services/Store/StoreProvider.tsx:349]: Will do before merge 👍

File: src/services/ApiService/History/types.ts (hunk 1 above)
Comment (FrederikBolding) [reply src/services/ApiService/History/types.ts:27]: We need to solve this in the API, reached out on Slack.

File: src/services/TxHistory/helpers.tsx (hunk 3 above)
Comment (FrederikBolding) [reply src/services/TxHistory/helpers.tsx:41]: Added test for `deriveTxType`

File: src/services/TxHistory/helpers.tsx (hunk 4 above)
Comment (FrederikBolding) [reply src/services/TxHistory/helpers.tsx:38]: No, you are right! 👍

File: src/services/ApiService/History/types.ts [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+  readonly hash: ITxHash;
+  readonly nonce: string; // Hex
+  readonly recipientAddress: TAddress;
+  readonly status: ITxStatus.PENDING | ITxStatus.SUCCESS | ITxStatus.FAILED | ITxStatus.UNKNOWN;
+  readonly timestamp?: number;
+  readonly to: TAddress;
+  readonly value: string; // Hex
+
+  readonly txType: ITxType;
+}
```
Comment (blurpesec) [inline src/services/ApiService/History/types.ts:20]: In my last pr i added these which i think would be good to stick here: https://github.com/MyCryptoHQ/MyCrypto/blob/master/src/types/transactionFlow.ts#L17-L23

**Review by blurpesec — APPROVED** [review]

**Pull request merged**